### PR TITLE
Add missing url in sync structs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

**Comment by frewsxcv** [comment]
```
std/sync/struct.PoisonError.html:62: broken link fragment  `#method.lock` pointing to `std/sync/struct.RwLock.html`

std/sync/type.TryLockResult.html:54: broken link - std/sync/struct.LockResult.html
```

File: src/libstd/sys_common/poison.rs [hunk 1]
```diff
@@ -60,10 +60,13 @@ pub struct Guard {
 
 /// A type of error which can be returned whenever a lock is acquired.
 ///
-/// Both Mutexes and RwLocks are poisoned whenever a thread fails while the lock
+/// Both [`Mutex`]es and [`RwLock`]s are poisoned whenever a thread fails while the lock
```
Comment (frewsxcv) [inline src/libstd/sys_common/poison.rs]: Maybe you meant https://doc.rust-lang.org/std/sync/struct.Condvar.html#method.wait

Comment (GuillaumeGomez) [inline src/libstd/sys_common/poison.rs]: Absolutely!

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**